### PR TITLE
#3760 Add another required() method with recommanded values into SettingDef

### PR DESCRIPTION
**Pull request by vitojeng**
required by #3760 

### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3760)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-common/src/main/java/com/island/ohara/common/setting/SettingDef.java [hunk 1]
```diff
@@ -722,6 +722,18 @@ public Builder required(Type valueType) {
       return checkAndSet(valueType, Necessary.REQUIRED, null);
     }
 
+    public Builder required(Set<String> recommendedValues) {
+      this.recommendedValues = Objects.requireNonNull(recommendedValues);
+      return checkAndSet(
+              Type.STRING, Necessary.REQUIRED, null);
+    }
+
+    public Builder required(String defaultValue, Set<String> recommendedValues) {
```
Comment (chia7712) [inline ohara-common/src/main/java/com/island/ohara/common/setting/SettingDef.java]: it is NOT required if there is default value (and it is duplicate to https://github.com/oharastream/ohara/blob/master/ohara-common/src/main/java/com/island/ohara/common/setting/SettingDef.java#L889)

File: ohara-configurator/src/main/scala/com/island/ohara/configurator/route/ObjectChecker.scala [hunk 2]
```diff
@@ -43,7 +43,7 @@ import scala.reflect.ClassTag
   * unify all resource checks and produces unified error message.
   */
 trait ObjectChecker {
-  def checkList: ChickList
+  def checkList: CheckList
```
Comment (chia7712) [inline ohara-configurator/src/main/scala/com/island/ohara/configurator/route/ObjectChecker.scala]: good :)

File: ohara-common/src/main/java/com/island/ohara/common/setting/SettingDef.java (hunk 1 above)
Comment (vitojeng) [reply ohara-common/src/main/java/com/island/ohara/common/setting/SettingDef.java]: Got it

**Comment by vitojeng** [comment]
retry manager-api

**Comment by vitojeng** [comment]
retry collieIT

**Comment by vitojeng** [comment]
retry connectorIT

**Pull request merged**